internal/agent/hooks/kcrypt.go

```diff
@@ -54,3 +74,46 @@ func (k Kcrypt) Run(c config.Config) error {
 
 	return nil
 }
+
+// updatePartitionLabelMapping takes partition information as a string argument
+// the the form: `label:name:uuid` (that's what the `kcrypt encrypt` command returns
+// on success. This function stores it in the PartitionInfoFile yaml file for
+// later use.
+func updatePartitionLabelMapping(partitionData, file string) error {
+	partitionInfo, err := parsePartitionInfoFile(file)
+	if err != nil {
+		return err
+	}
+
+	parts := strings.Split(partitionData, ":")
+	partitionInfo[parts[0]] = parts[2]
+
+	return updatePartitionInfoFile(partitionInfo, file)
+}
+
+func parsePartitionInfoFile(file string) (PartitionInfo, error) {
```

Comment:
This is all good, nice! I'm just wondering here - as the challenger will need to re-read the same file, if makes sense to make those public in a separate package dedicated for it instead

Reply:
Moved to kcrypt: https://github.com/kairos-io/kcrypt/pull/3/commits/82c6e8fcd092bd05aeda8fd5ef292965957ee95f

I'll import it here as a module.